pkgs/build-support/rust/build-rust-crate.nix

```diff
@@ -297,9 +297,15 @@ in
 crate_: lib.makeOverridable ({ rust, release, verbose, features, buildInputs, crateOverrides }:
 
 let crate = crate_ // (lib.attrByPath [ crate_.crateName ] (attr: {}) crateOverrides crate_);
+    processedAttrs = [
+      "src" "buildInputs" "crateBin" "crateLib" "libName" "libPath"
+      "buildDependencies" "dependencies" "features"
+      "crateName" "version" "build" "authors" "colors"
+    ];
+    extraDerivationAttrs = lib.filterAttrs (n: v: ! lib.elem n processedAttrs) crate;
```

Comment:
Rather then filtering I would make an `assertion` to make debugging easier, in case somebody tries to set them.
Or is there any use case for that?

Reply:
The intent here was to remove only the attributes that had a special interpretation in the body below. All of these are valid to set. Re-reading, I think I could simplify colors, build, libName and libPath.

Reply:
Yes, I see that. But users might try to override them in `cratesOverrides` and wonder why the override has no effect.

Reply:
When users set these fields in `crateOverrides`, the overrides become part of `crate`, and will have some meaning in the body. They're not being silently ignored. I agree it's potentially confusing that setting `crateBin` doesn't pass through without being transformed, or that `features` is renamed and transformed to `crateFeatures`, but they do make it through in some expression.

Reply:
ok. Then we it should be kept this way.